akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -258,6 +258,15 @@ object Sink {
   def foreach[T](f: T ⇒ Unit): Sink[T, Future[Done]] =
     Flow[T].map(f).toMat(Sink.ignore)(Keep.right).named("foreachSink")
 
+  /**
+   * A `Sink` that will invoke the given procedure asynchronously for each received element. The sink is materialized
+   * into a [[scala.concurrent.Future]] which will be completed with `Success` when reaching the
+   * normal end of the stream, or completed with `Failure` if there is a failure signaled in
+   * the stream.
+   */
+  def foreachAsync[T](parallelism: Int)(f: T ⇒ Future[Unit]): Sink[T, Future[Done]] =
+    Flow[T].mapAsync(parallelism)(f).toMat(Sink.ignore)(Keep.right).named("foreachAsyncSink")
```

Comment:
this should be `mapAsyncUnordered` since the order of the returned Unit values don't matter

Reply:
What about the sink processing? Depending, one can be worried about the order. Maybe also create a `foreachAsyncUnordered`?

Reply:
the `f` will be called in order, so I don't see the problem